diff --git a/arch/alpha/config.in b/arch/alpha/config.in
index e2b68bf..ea233e0 100644
--- a/arch/alpha/config.in
+++ b/arch/alpha/config.in
@@ -49,10 +49,13 @@ fi
 bool 'Networking support' CONFIG_NET
 bool 'System V IPC' CONFIG_SYSVIPC
 tristate 'Kernel support for ELF binaries' CONFIG_BINFMT_ELF
+endmenu
+
 
 mainmenu_option next_comment
 comment 'Loadable module support'
 bool 'Set version information on all symbols for modules' CONFIG_MODVERSIONS
+endmenu
 
 source drivers/block/Config.in
 
@@ -68,6 +71,7 @@ tristate 'SCSI support' CONFIG_SCSI
 if [ "$CONFIG_SCSI" != "n" ]; then
   source drivers/scsi/Config.in
 fi
+endmenu
 
 if [ "$CONFIG_NET" = "y" ]; then
   mainmenu_option next_comment
@@ -77,6 +81,7 @@ if [ "$CONFIG_NET" = "y" ]; then
   if [ "$CONFIG_NETDEVICES" = "y" ]; then
     source drivers/net/Config.in
   fi
+  endmenu
 fi
 
 mainmenu_option next_comment
@@ -86,6 +91,7 @@ bool 'Support non-SCSI/IDE/ATAPI drives' CONFIG_CD_NO_IDESCSI
 if [ "$CONFIG_CD_NO_IDESCSI" != "n" ]; then
   source drivers/cdrom/Config.in
 fi
+endmenu
 
 source fs/Config.in
 
@@ -99,6 +105,8 @@ if [ "$CONFIG_SOUND" != "n" ]; then
   source drivers/sound/Config.in
 fi
 
+endmenu
+
 mainmenu_option next_comment
 comment 'Kernel hacking'
 
diff --git a/arch/i386/config.in b/arch/i386/config.in
index 4ae35e1..c2804b7 100644
--- a/arch/i386/config.in
+++ b/arch/i386/config.in
@@ -23,10 +23,12 @@ fi
 #if [ "$CONFIG_M586" = "n" ]; then
 bool 'Use -m486 flag for 486-specific optimizations' CONFIG_M486
 #fi
+endmenu
 
 mainmenu_option next_comment
 comment 'Loadable module support'
 bool 'Set version information on all symbols for modules' CONFIG_MODVERSIONS
+endmenu
 
 source drivers/block/Config.in
 
@@ -42,6 +44,7 @@ tristate 'SCSI support' CONFIG_SCSI
 if [ "$CONFIG_SCSI" != "n" ]; then
   source drivers/scsi/Config.in
 fi
+endmenu
 
 if [ "$CONFIG_NET" = "y" ]; then
   mainmenu_option next_comment
@@ -51,6 +54,7 @@ if [ "$CONFIG_NET" = "y" ]; then
   if [ "$CONFIG_NETDEVICES" = "y" ]; then
     source drivers/net/Config.in
   fi
+  endmenu
 fi
 
 mainmenu_option next_comment
@@ -60,6 +64,7 @@ bool 'Support non-SCSI/IDE/ATAPI drives' CONFIG_CD_NO_IDESCSI
 if [ "$CONFIG_CD_NO_IDESCSI" != "n" ]; then
   source drivers/cdrom/Config.in
 fi
+endmenu
 
 source fs/Config.in
 
@@ -72,6 +77,7 @@ tristate 'Sound card support' CONFIG_SOUND
 if [ "$CONFIG_SOUND" != "n" ]; then
   source drivers/sound/Config.in
 fi
+endmenu
 
 mainmenu_option next_comment
 comment 'Kernel hacking'
diff --git a/arch/sparc/config.in b/arch/sparc/config.in
index d1cedda..5041eb7 100644
--- a/arch/sparc/config.in
+++ b/arch/sparc/config.in
@@ -21,6 +21,7 @@ bool 'Networking support' CONFIG_NET n
 bool 'Limit memory to low 16MB' CONFIG_MAX_16M n
 bool 'System V IPC' CONFIG_SYSVIPC y
 bool 'Kernel support for ELF binaries' CONFIG_BINFMT_ELF y
+endmenu
 
 if [ "$CONFIG_NET" = "y" ]; then
 mainmenu_option next_comment
@@ -39,6 +40,7 @@ bool 'Disable NAGLE algorithm (normally enabled)' CONFIG_TCP_NAGLE_OFF n
 fi
 bool 'The IPX protocol' CONFIG_IPX n
 #bool 'Amateur Radio AX.25 Level 2' CONFIG_AX25 n
+endmenu
 fi
 
 mainmenu_option next_comment
@@ -64,6 +66,7 @@ comment 'SCSI low-level drivers'
 bool 'Sun ESP Scsi support' CONFIG_SCSI_SUN_ESP n
 
 fi
+endmenu
 
 
 if [ "$CONFIG_NET" = "y" ]; then
@@ -89,6 +92,7 @@ bool 'Load balancing support (experimental)' CONFIG_SLAVE_BALANCING n
 bool 'Sun LANCE Ethernet support' CONFIG_SUN_LANCE n
 bool 'Sun Intel Ethernet support' CONFIG_SUN_INTEL n
 fi
+endmenu
 fi
 
 mainmenu_option next_comment
@@ -110,17 +114,19 @@ else
 fi
 bool 'OS/2 HPFS filesystem support (read only)' CONFIG_HPFS_FS n
 bool 'System V and Coherent filesystem support' CONFIG_SYSV_FS n
-
+endmenu
 
 mainmenu_option next_comment
 comment 'character devices'
 
 bool 'Zilog serial support' CONFIG_SUN_ZS n
+endmenu
 
 comment 'Sound'
 
 bool 'Sun Audio support' CONFIG_SUN_AUDIO n
 
+
 mainmenu_option next_comment
 comment 'Kernel hacking'
 
@@ -131,4 +137,4 @@ if [ "$CONFIG_PROFILE" = "y" ]; then
 fi
 if [ "$CONFIG_SCSI" = "y" ]; then
 bool 'Verbose SCSI error reporting (kernel size +=12K)' CONFIG_SCSI_CONSTANTS y
-fi
+fi
\ No newline at end of file
