diff --git a/arch/sparc/config.in b/arch/sparc/config.in
index 9829bbb..bf67e66 100644
--- a/arch/sparc/config.in
+++ b/arch/sparc/config.in
@@ -48,6 +48,7 @@ if [ "$CONFIG_BLK_DEV_RAM" = "y" ]; then
 fi
 
 tristate 'Loopback device support' CONFIG_BLK_DEV_LOOP
+endmenu
 
 if [ "$CONFIG_NET" = "y" ]; then
 	source net/Config.in
@@ -76,6 +77,7 @@ if [ "$CONFIG_SCSI" != "n" ]; then
 	comment 'SCSI low-level drivers'
 
 	dep_tristate 'Sparc ESP Scsi Driver' CONFIG_SCSI_SUNESP $CONFIG_SCSI
+	endmenu
 fi
 endmenu
 
