diff --git a/arch/mips/config.in b/arch/mips/config.in
index aa1aba3..765ece6 100644
--- a/arch/mips/config.in
+++ b/arch/mips/config.in
@@ -217,7 +217,7 @@ if [ "$CONFIG_SGI" != "y" -a "$CONFIG_DECSTATION" != "y" -a "$CONFIG_BAGET_MIPS"
 
 
   mainmenu_option next_comment
-  comment comment 'Old CD-ROM drivers (not SCSI, not IDE)'
+  comment 'Old CD-ROM drivers (not SCSI, not IDE)'
 
   bool 'Support non-SCSI/IDE/ATAPI drives' CONFIG_CD_NO_IDESCSI
   if [ "$CONFIG_CD_NO_IDESCSI" != "n" ]; then
