diff --git a/arch/arm/config.in b/arch/arm/config.in
index af7b3fc..27b8e82 100644
--- a/arch/arm/config.in
+++ b/arch/arm/config.in
@@ -145,7 +145,7 @@ if [ "$CONFIG_ARCH_ACORN" = "y" ]; then
 fi
 
 if [ "$CONFIG_VGA_CONSOLE" = "n" -a "$CONFIG_FB" = "n" ]; then
-  source arch/arm/drivers/char/Config.in
+  #source arch/arm/drivers/char/Config.in
 else
   source drivers/char/Config.in
 fi
diff --git a/arch/mips/config.in b/arch/mips/config.in
index b5d7916..2d02322 100644
--- a/arch/mips/config.in
+++ b/arch/mips/config.in
@@ -207,7 +207,7 @@ fi
 endmenu
 
 if [ "$CONFIG_SGI" = "y" ]; then
-  source drivers/sgi/char/Config.in
+  #source drivers/sgi/char/Config.in
 fi
 
 mainmenu_option next_comment
