diff --git a/arch/arm/config.in b/arch/arm/config.in
index ed7135a..c9b75c7 100644
--- a/arch/arm/config.in
+++ b/arch/arm/config.in
@@ -145,7 +145,7 @@ if [ "$CONFIG_ARCH_ACORN" = "y" ]; then
 fi
 
 if [ "$CONFIG_VGA_CONSOLE" = "n" -a "$CONFIG_FB" = "n" ]; then
-  source arch/arm/drivers/char/Config.in
+  #source arch/arm/drivers/char/Config.in
 else
   source drivers/char/Config.in
 fi
diff --git a/arch/mips/config.in b/arch/mips/config.in
index b4c92b4..0b6a63a 100644
--- a/arch/mips/config.in
+++ b/arch/mips/config.in
@@ -217,7 +217,7 @@ if [ "$CONFIG_SGI" != "y" -a "$CONFIG_DECSTATION" != "y" -a "$CONFIG_BAGET_MIPS"
 
 
   mainmenu_option next_comment
-  comment comment 'Old CD-ROM drivers (not SCSI, not IDE)'
+  comment 'Old CD-ROM drivers (not SCSI, not IDE)'
 
   bool 'Support non-SCSI/IDE/ATAPI drives' CONFIG_CD_NO_IDESCSI
   if [ "$CONFIG_CD_NO_IDESCSI" != "n" ]; then
