diff --git a/arch/arm/config.in b/arch/arm/config.in
index 544f39b..03cbe91 100644
--- a/arch/arm/config.in
+++ b/arch/arm/config.in
@@ -579,7 +579,7 @@ fi
 endmenu
 
 if [ "$CONFIG_ARCH_CLPS711X" = "y" ]; then
-   source drivers/ssi/Config.in
+   #source drivers/ssi/Config.in
 fi
 
 source drivers/ieee1394/Config.in
diff --git a/arch/sh/config.in b/arch/sh/config.in
index 136513e..a15f692 100644
--- a/arch/sh/config.in
+++ b/arch/sh/config.in
@@ -263,7 +263,7 @@ endmenu
 source drivers/input/Config.in
 
 if [ "$CONFIG_SH_DREAMCAST" = "y" ]; then
-   source drivers/maple/Config.in
+   #source drivers/maple/Config.in
 fi
 
 mainmenu_option next_comment
@@ -307,7 +307,7 @@ if [ "$CONFIG_SH_DREAMCAST" = "y" -a "$CONFIG_MAPLE" != "n" ]; then
    endmenu
 fi
 
-source drivers/char/joystick/Config.in
+#source drivers/char/joystick/Config.in
 
 if [ "$CONFIG_PARPORT" != "n" ]; then
    dep_tristate 'Parallel printer support' CONFIG_PRINTER $CONFIG_PARPORT
