diff --git a/arch/arm/config.in b/arch/arm/config.in
index 8459184..cab823b 100644
--- a/arch/arm/config.in
+++ b/arch/arm/config.in
@@ -494,7 +494,7 @@ fi
 endmenu
 
 if [ "$CONFIG_ARCH_CLPS711X" = "y" ]; then
-   source drivers/ssi/Config.in
+   #source drivers/ssi/Config.in
 fi
 
 source drivers/ieee1394/Config.in
diff --git a/arch/cris/drivers/Config.in b/arch/cris/drivers/Config.in
index 7d934a6..f1a6754 100644
--- a/arch/cris/drivers/Config.in
+++ b/arch/cris/drivers/Config.in
@@ -178,7 +178,7 @@ fi
 bool 'ARTPEC-1 support' CONFIG_JULIETTE
 
 if [ "$CONFIG_JULIETTE" = "y" ]; then
-   source arch/cris/drivers/juliette/Config.in
+   #source arch/cris/drivers/juliette/Config.in
 fi
 
 bool 'USB host' CONFIG_ETRAX_USB_HOST
diff --git a/arch/sh/config.in b/arch/sh/config.in
index 9859d31..cb3046d 100644
--- a/arch/sh/config.in
+++ b/arch/sh/config.in
@@ -281,7 +281,7 @@ endmenu
 source drivers/input/Config.in
 
 if [ "$CONFIG_SH_DREAMCAST" = "y" ]; then
-   source drivers/maple/Config.in
+   #source drivers/maple/Config.in
 fi
 
 mainmenu_option next_comment
diff --git a/drivers/s390/Config.in b/drivers/s390/Config.in
index f3b8963..de14446 100644
--- a/drivers/s390/Config.in
+++ b/drivers/s390/Config.in
@@ -17,18 +17,18 @@ if [ "$CONFIG_DASD" != "n"  ]; then
   dep_tristate '   Support for ECKD Disks' CONFIG_DASD_ECKD $CONFIG_DASD
   if [ "$CONFIG_DASD_ECKD" = "m" ]; then
     bool     '   Automatic activation of ECKD module' CONFIG_DASD_AUTO_ECKD
-  fi;
+  fi
   dep_tristate '   Support for FBA  Disks' CONFIG_DASD_FBA $CONFIG_DASD
   if [ "$CONFIG_DASD_FBA" = "m" ]; then
     bool     '   Automatic activation of FBA  module' CONFIG_DASD_AUTO_FBA
-  fi;
+  fi
 #  dep_tristate '   Support for CKD  Disks' CONFIG_DASD_CKD $CONFIG_DASD
   if [ "$CONFIG_ARCH_S390X" != "y" ]; then
     dep_tristate '   Support for DIAG access to CMS reserved Disks' CONFIG_DASD_DIAG $CONFIG_DASD
     if [ "$CONFIG_DASD_DIAG" = "m" ]; then
       bool     '   Automatic activation of DIAG module' CONFIG_DASD_AUTO_DIAG
-    fi;
-  fi; 
+    fi
+  fi
 fi
 
 endmenu
