diff --git a/arch/arm/config.in b/arch/arm/config.in
index 635874b..5bf116c 100644
--- a/arch/arm/config.in
+++ b/arch/arm/config.in
@@ -466,7 +466,7 @@ fi
 endmenu
 
 if [ "$CONFIG_ARCH_CLPS711X" = "y" ]; then
-   source drivers/ssi/Config.in
+   #source drivers/ssi/Config.in
 fi
 
 source drivers/ieee1394/Config.in
diff --git a/arch/sh/config.in b/arch/sh/config.in
index 54640eb..d304b84 100644
--- a/arch/sh/config.in
+++ b/arch/sh/config.in
@@ -262,7 +262,7 @@ endmenu
 source drivers/input/Config.in
 
 if [ "$CONFIG_SH_DREAMCAST" = "y" ]; then
-   source drivers/maple/Config.in
+   #source drivers/maple/Config.in
 fi
 
 mainmenu_option next_comment
@@ -306,7 +306,7 @@ if [ "$CONFIG_SH_DREAMCAST" = "y" -a "$CONFIG_MAPLE" != "n" ]; then
    endmenu
 fi
 
-source drivers/char/joystick/Config.in
+#source drivers/char/joystick/Config.in
 
 if [ "$CONFIG_PARPORT" != "n" ]; then
    dep_tristate 'Parallel printer support' CONFIG_PRINTER $CONFIG_PARPORT
diff --git a/arch/sparc/config.in b/arch/sparc/config.in
index cd1e059..65e224b 100644
--- a/arch/sparc/config.in
+++ b/arch/sparc/config.in
@@ -226,7 +226,7 @@ tristate 'Sound card support' CONFIG_SOUND
 if [ "$CONFIG_SOUND" != "n" ]; then
    source sound/Config.in
 fi
-endmenu
+
 
 source drivers/usb/Config.in
 
