diff --git a/drivers/hwmon/Kconfig b/drivers/hwmon/Kconfig
index 68cf8774..d1a33fc0 100644
--- a/drivers/hwmon/Kconfig
+++ b/drivers/hwmon/Kconfig
@@ -334,7 +334,7 @@ config SENSORS_F71882FG
 	  will be called f71882fg.
 
 config SENSORS_F75375S
-	tristate "Fintek F75375S/SP and F75373";
+	tristate "Fintek F75375S/SP and F75373"
 	depends on I2C && EXPERIMENTAL
 	help
 	  If you say yes here you get support for hardware monitoring
