diff --git a/arch/arm/mach-sa1100/Kconfig b/arch/arm/mach-sa1100/Kconfig
index b586220d..fe6bce95 100644
--- a/arch/arm/mach-sa1100/Kconfig
+++ b/arch/arm/mach-sa1100/Kconfig
@@ -248,7 +248,7 @@ config SA1100_SIMPAD
 config SA1100_PFS168
 	bool "Tulsa"
 	help
-	  The Radisys Corp. PFS-168 (aka Tulsa) is an Intel® StrongArm® SA-1110 based
+	  The Radisys Corp. PFS-168 (aka Tulsa) is an IntelÂ® StrongArmÂ® SA-1110 based
 	  computer which includes the SA-1111 Microprocessor Companion Chip and other
 	  custom I/O designed to add connectivity and multimedia features for vending
 	  and business machine applications. Say Y here if you require support for
diff --git a/drivers/mtd/maps/Kconfig b/drivers/mtd/maps/Kconfig
index cdc5bbd1..30aca55d 100644
--- a/drivers/mtd/maps/Kconfig
+++ b/drivers/mtd/maps/Kconfig
@@ -335,7 +335,7 @@ config MTD_CFI_FLAGADM
 	tristate "CFI Flash device mapping on FlagaDM"
 	depends on PPC32 && 8xx && MTD_CFI
 	help
-	  Mapping for the Flaga digital module. If you don´t have one, ignore
+	  Mapping for the Flaga digital module. If you don't have one, ignore
 	  this setting.
 
 config MTD_BEECH
