diff --git a/arch/cris/Kconfig b/arch/cris/Kconfig
index 222da150..7bbac0b7 100644
--- a/arch/cris/Kconfig
+++ b/arch/cris/Kconfig
@@ -158,7 +158,8 @@ source "net/Kconfig"
 # bring in ETRAX built-in drivers
 menu "Drivers for built-in interfaces"
 # arch/cris/arch is a symlink to correct arch (arch-v10 or arch-v32)
-source arch/cris/arch/drivers/Kconfig
+source arch/cris/arch-v10/drivers/Kconfig
+source arch/cris/arch-v32/drivers/Kconfig
 
 endmenu
 
diff --git a/arch/um/Kconfig b/arch/um/Kconfig
index dd1689b8..93317aae 100644
--- a/arch/um/Kconfig
+++ b/arch/um/Kconfig
@@ -81,7 +81,7 @@ config STATIC_LINK
 	  Additionally, this option enables using higher memory spaces (up to
 	  2.75G) for UML.
 
-source "arch/um/Kconfig.arch"
+#source "arch/um/Kconfig.arch"
 source "mm/Kconfig"
 source "kernel/time/Kconfig"
 
diff --git a/drivers/hwmon/Kconfig b/drivers/hwmon/Kconfig
index a0445bea..0a1267a1 100644
--- a/drivers/hwmon/Kconfig
+++ b/drivers/hwmon/Kconfig
@@ -248,7 +248,7 @@ config SENSORS_F71882FG
 	  will be called f71882fg.
 
 config SENSORS_F75375S
-	tristate "Fintek F75375S/SP and F75373";
+	tristate "Fintek F75375S/SP and F75373"
 	depends on I2C && EXPERIMENTAL
 	help
 	  If you say yes here you get support for hardware monitoring
